### PR TITLE
1.9.0: More plain Result features

### PR DESCRIPTION
**Pull request by JesusMcCloud**
- add WasmJS target
- add WasmWasi target
- add `wrappingPlain`, which works just like `wrapping` but on a `Result` rather than a `KmmResult` to avoid instantiation overhead
- rename `catchingUnwrapped` to `catchingPlain` to avoid confusions with `wrapping` but keep the old name as deprecated alternative
- add `nonFatalOrThrow`

### CODE REVIEW

**Review by iaik-jheher — CHANGES_REQUESTED** [review]
I'd suggest not renaming `catchingUnwrapped`, but renaming `wrapping`.
`catchingUnwrapped` matches `KmmResult<*>.unwrap(): Result<*>` and its inverse, `Result<*>.wrap(): KmmResult<*>`. This means that `wrapping` is the odd one out.

I'd suggest (the type parameter indicates the contained failure case, not the result type):
* `catching {...} -> KmmResult<*>`
* `catchingAs(a=::T) {...} -> KmmResult<out T>`
* `catchingUnwrapped {...} ->Result<*>`
* `catchingUnwrappedAs(a=::T) {...} -> Result<out T>`

In other words, `wrapping` becomes `catchingAs`, and the new `wrappingPlain` becomes `catchingUnwrappedAs`.

File: README.md [hunk 1]
```diff
@@ -88,12 +87,12 @@ KmmResult comes with `catching`. This is a non-fatal-only-catching version of st
...
 
-The only downside of `catching` is that it incurs instatiation overhead, because it creates a `KmmResult` instance.
+The only downside of `catching` is that it incurs instantiation overhead, because it creates a `KmmResult` instance.
 Internally, though, only the behaviour is important, not Swift interop. Hence, you don't care for a `KmmResult` and you
 certainly don't care for the cost of instantiating an object. Here, the `Result.nonFatalOrThrow()` extension shipped with KmmResult
 comes to the rescue. It does exactly what the name suggest: It re-throws any fatal exception and leaved the `Result` object
-untouched otherwise.
-
+untouched otherwise.  
+In addition, there's `catchingPlain` which directly returns an stdlib `Result`
```
[the remark this answers is not in the document]
Comment (JesusMcCloud) [reply README.md:95]: also, its not called that anymore…

[the remark this answers is not in the document]
Comment (iaik-jheher) [reply README.md:95]: yes, also that :)

**Comment by iaik-jheher** [comment]
@JesusMcCloud ready for review

**Comment by iaik-jheher** [comment]
track [KT-72600](https://youtrack.jetbrains.com/issue/KT-72600/Contracts-are-not-transitive) for contract boilerplate

**Comment by iaik-jheher** [comment]
track [KT-72601](https://youtrack.jetbrains.com/issue/KT-72601/LowPriorityInOverloadResolution-needed-for-constructor-reference-disambiguation) for `@Suppress` hackery

**Comment by JesusMcCloud** [comment]
will release on Monday evening, unless someone objects

File: CHANGELOG.md [hunk 2]
```diff
@@ -55,4 +55,11 @@
   - `result.shouldSucceed()` returning the contained value
 - remove Arrow dependency and import arrow's list of Fatal exceptions directly into our code
 - Introduce `Result.nonFatalOrThrow` to mimic KmmResult's non-fatal-only behaviour, but without the object instantiation overhead
-- Introduce `carchingUnwrapped`, which mimics KmmResult's non-fatal-only behaviour, but without the object instantiation overhead
\ No newline at end of file
+- Introduce `carchingUnwrapped`, which mimics KmmResult's non-fatal-only behaviour, but without the object instantiation overhead
+
+## 1.9.0
+- add WasmJS target
+- add WasmWasi target (not or KmmResult-test, as Kotest does not support WASI yet)
```
Comment (nodh) [inline CHANGELOG.md:62]: Typo in `or`?

**Review by nodh — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (iaik-jheher) [inline CHANGELOG.md:62]:
```suggestion
- add WasmWasi target (not for KmmResult-test, as Kotest does not support WASI yet)
```

**Review by iaik-jheher — APPROVED** [review]
